### PR TITLE
Opt-in support for sign-extend

### LINKED ISSUE #66: Add support for sign-extension instructions (post MVP feature)
Hello,

I'm opening this issue to report an error I ran into where it appeared wazero was unable to instantiate an AssemblyScript program compiled with asc.

Here is the error I got:
```
functions: invalid function at index 106/280: invalid instruction
```

Tools like `wasm2wat` are able to read through the file without issues, so it appears correct tho I do not know whether they perform checks similar to those in wazero.

I'm attaching the program that triggered the error:
[test.wasm.gz](https://github.com/tetratelabs/wazero/files/7674600/test.wasm.gz)

Let me know if you need any other information.

### PR DESCRIPTION
**Pull request by mathetake**
Allows users to do the following to enable sign-extension-ops

```
r := wazero.NewRuntimeWithConfig(wazero.NewRuntimeConfig().WithFeatureSignExtensionOps(true))
```

Resolves #66

### CODE REVIEW

File: internal/wasm/interpreter/interpreter_test.go [hunk 1]
```diff
@@ -117,3 +120,115 @@ func TestEngine_Call_HostFn(t *testing.T) {
 		require.EqualError(t, err, "expected 1 params, but passed 2")
 	})
 }
+
+func TestCallEngine_callNativeFunc_signExtend(t *testing.T) {
+	v := uint64(0xfedcba98_80000000)
+	fmt.Println(int64(v))
```
Comment (mathetake) [inline internal/wasm/interpreter/interpreter_test.go:126]:
```suggestion
```

**Comment by codefromthecrypt** [comment]
going to add text compiler support so that this PR can be an examplar

**Comment by mathetake** [comment]
nice

**Review by codefromthecrypt — APPROVED** [review]
Great

**Pull request merged**